### PR TITLE
#273 Do not remove products

### LINKED ISSUE #273: Похожие товары на 404 странице
На 404 странице под кнопкой "На главную" выводим  10 рандомных товаров их то же категории, где был удаленный товар. 
С заголовком "Похожие товары"
Две строки по 5 товаров. как на главной http://prntscr.com/ivclw1

### PR DESCRIPTION
**Pull request by artemiy312**
Closes #273 

Prevent products removing during 1C update

### CODE REVIEW

**Comment by duker33** [comment]
@artemiy312 , it's not resolve issue. Leave pdd subtasks for 404 plz

**Comment by artemiy312** [comment]
@duker33 I already have [created it](https://github.com/fidals/shopelectro/pull/443/files#diff-9e53a25fb3c9eb1745ff37a5799ef9daR57)

File: shopelectro/management/commands/excel.py [hunk 1]
```diff
@@ -112,7 +112,7 @@ def collapse_row(row):
         def write_product_rows():
             """Write products lines."""
             sheet = self.sheet
-            products = Product.objects.filter(category=category)
+            products = Product.objects.filter(category=category, page__is_active=True)
```
Comment (duker33) [inline shopelectro/management/commands/excel.py]: let's add custom ModelManager instead:

```
class ProductManager(models.Manager):
    def get_queryset(self):
        return super(DahlBookManager, self).get_queryset().filter(author='Roald Dahl')

class Product:
    active = ProductManager()

# in client code
Product.active.filter(category=category)
```

Took example here: https://docs.djangoproject.com/en/1.11/topics/db/managers/#modifying-a-manager-s-initial-queryset

File: shopelectro/views/catalog.py [hunk 2]
```diff
@@ -45,11 +45,22 @@ class ProductPage(catalog.ProductPage):
...
         .select_related('page')
     )
 
+    def get(self, request, *args, **kwargs):
+        try:
+            self.object = self.get_object()
+        except http.Http404 as error404:
+            # @todo #273 Render 404, that is recommending products for a deleted product.
+            #  1. Find a product with page__is_active=False
+            #  2. If the product exists then render 404 with products of its category
```
Comment (duker33) [inline shopelectro/views/catalog.py]: Let's add this:
```
...
3. See parent task for details.
```

Some details about design are missed in this task body

File: shopelectro/management/commands/excel.py (hunk 1 above)
Comment (artemiy312) [reply shopelectro/management/commands/excel.py]: @duker33 I am going to create todo for it and it seems that this code will not work, but we may discuss it in the related issue

**Pull request merged**